compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -1254,7 +1254,10 @@ class Namer { typer: Typer =>
               newSymbol(cls, forwarderName, mbrFlags, mbrInfo, coord = span)
 
           forwarder.info = avoidPrivateLeaks(forwarder)
-          forwarder.addAnnotations(sym.annotations.filterConserve(_.symbol != defn.BodyAnnot))
+          forwarder.addAnnotations(sym.annotations.filterConserve { annot =>
+            annot.symbol != defn.BodyAnnot
+            && annot.symbol != defn.TailrecAnnot
```

Comment:
alternatively, could the tailrec check for the exported flag and ignore it?

Reply:
Right, we also need to filter macro annotations. 

I think that disabling the behavior where it is checked will be less scalable. For example, I disable the macro annotations on exports I have to find one or more places where this restriction needs to be taken into account. And those places are usually code generation details like exports.

Reply:
I also added `@main`. Might might still need to filter others.

Reply:
I wonder if we should have a whitelist approach with this one. This would make sure that we never accidentally keep an annotation we do not want.

I tested this on `scala3-bootstrapped/testCompilation` and it only required the `@targetName` annotation.

Reply:
The thing is that we will never know what should be the behavior for custom annotations.

The only holistic solution here is to have an `@meta`-based indication of what to do. This is the approach used to determine whether annotations should go to fields/getters/setters. It would also be appropriate to determine whether it should go to exports/bridges/etc.

Reply:
Maybe we need `ExportableAnnotation` trait that all such annotations should extend to get exported.